src/cpuid/src/transformer/intel.rs

```diff
@@ -20,7 +20,7 @@ fn update_deterministic_cache_entry(
     // Put all the cores in the same socket
     entry.eax.write_bits_in_range(
         &eax::MAX_CORES_PER_PACKAGE_BITRANGE,
-        u32::from(vm_spec.cpu_count - 1),
+        u32::from(vm_spec.cpu_count / vm_spec.cpus_per_core()) - 1,
```

Comment:
This doesnt seem right. Why does HT matter here?

Reply:
This represents the number of cores that share the L3 cache. For example:
1. If there are 64 vCPUs with HT in a socket, then there are 32 cores that share the L3 cache
2. If there are 64 vCPUs without HT in a socket, then there are 64 actual cores that share the L3 cache

Reply:
I did some memory performance tests using the `STREAM` benchmark. The results after this change look quite similar to the ones before:

```
8 vCPUs + HT enabled

master:

RUN: 100
Copy:	68109.10    72917.20    70080.12    6.59%
Scale:	61081.00    63133.30    62091.64    3.25%
Add:	69918.90    73565.80    71819.58    4.96%
Triad:	69215.70    71782.20    70491.32    3.58%

increase_vcpu_count:

RUN: 100
Copy:	68442.50    73089.70    70399.71    6.36%
Scale:	61176.50    63119.70    62015.29    3.08%
Add:	70422.10    73581.20    71745.81    4.29%
Triad:	69582.00    71878.90    70381.15    3.20%
```

The columns represent in oredr: min, max, average, deviation between min and max.

I will also try with 4 vCPUs, but I don't expect to see changes.

Reply:
The results for 4 vCPUs + HT enabled:
```
master:

RUN: 100
Copy:	40608.80    41110.60    40860.20    1.22%
Scale:	46015.40    46683.00    46375.26    1.43%
Add:	50238.30    50878.20    50556.40    1.26%
Triad:	50491.80    51124.80    50800.57    1.24%

increase_vcpu_count:

RUN: 100
Copy:	40549.90    41182.60    40823.37    1.54%
Scale:	45698.50    46753.70    46315.58    2.26%
Add:	50125.40    51033.00    50553.19    1.78%
Triad:	50432.50    51095.20    50768.45    1.30%
```